### PR TITLE
Check frame dimensions, data type and compression against dataset

### PR DESCRIPTION
**Pull request by GDYendell**
Error raised if values are set and do not a match the definition
for the dataset named by the frame.

### CODE REVIEW

File: frameProcessor/src/FileWriterPlugin.cpp [hunk 1]
```diff
@@ -811,6 +847,9 @@ void FileWriterPlugin::configureDataset(OdinData::IpcMessage& config, OdinData::
         dset_def.compression = config.get_param<std::string>(FileWriterPlugin::CONFIG_DATASET_COMPRESSION);
         LOG4CXX_DEBUG(logger_, "Enabling compression: " << dset_def.compression);
       }
+      else {
+        dset_def.compression = "none";
```
Comment (MattTaylorDLS) [inline frameProcessor/src/FileWriterPlugin.cpp]: Can we make this a constant somewhere?

File: frameProcessor/include/Frame.h [hunk 2]
```diff
@@ -120,6 +124,10 @@ class Frame
   unsigned long long frameNumber_;
   /** Map of dimensions, indexed by name */
   std::map<std::string, dimensions_t> dimensions_;
+  /** Compression type of raw data */
+  std::string compression_;
+  /** Compression type of raw data */
```
Comment (MattTaylorDLS) [inline frameProcessor/include/Frame.h]: comment wrong

File: frameProcessor/src/Frame.cpp [hunk 3]
```diff
@@ -179,6 +181,26 @@ dimensions_t Frame::get_dimensions(const std::string& type) const
...
+ * \return compression type.
+ */
+std::string Frame::get_compression() const
+{
+  return compression_;
+}
+
+/** Retrieves the data type of the raw data.
+ *
+ * -1: Unset 0: UINT8, 1: UINT16, 2: UINT32
```
Comment (MattTaylorDLS) [inline frameProcessor/src/Frame.cpp]: comma needed after Unset

**Comment by GDYendell** [comment]
Frame has no concept of those enums currently; they are in the FileWriterPlugin. They could be copied to the Frame so it can return the correct types. Or maybe we should create a common place to define them.

**Comment by ajgdls** [comment]
OK, I think the enums should be present in the frame as the frame now contains solid implementations that make use of them.  The file writer happens to also want to use them but it no longer owns them in my opinion.  Any class that uses frame object may or may not need to use those methods and so I think the frame should own the enums.
Perhaps this is another task that should be in a separate PR?

**Comment by GDYendell** [comment]
Sounds reasonable. If we are doing that can we rename PixelType and its enum values to something more suitable for a property of a Frame; maybe DataType, or BitDepth?

**Comment by ajgdls** [comment]
Sure that is fine with me.  So I'll approve this PR, we can create another for the follow up items.

**Pull request merged**